internal/clients/response.go

```diff
@@ -11,5 +13,9 @@ import (
 // request, and returns true if the response code was 404 not found
 func IsResponseCodeNotFound(err error) bool {
 	var apiErr *runtime.APIError
-	return errors.As(err, &apiErr) && apiErr.Code == http.StatusNotFound
+	if errors.As(err, &apiErr) {
+		return apiErr.Code == http.StatusNotFound
+	} else {
+		return strings.Contains(err.Error(), fmt.Sprintf("[%d]", http.StatusNotFound))
```

Comment:
This is interesting, did you come across a case that required this condition? I would be slightly wary about changing the behavior of this function because a lot of resources use it to determine if they exist or not. Another wrinkle is that the format of response error messages may change with the work @bcmdarroch is looking at.

Reply:
So this is a bit of a bandaid until I get the rest of service SDKs wired up with errors. Once https://github.com/hashicorp/hcp-sdk-go/pull/7 merges, I'd like to bump the SDK version, then upgrade the provider and revisit this helper then.

Reply:
Right, the "not found consul cluster" errors were not being marshalled into an apiErr obj. As @bcmdarroch mentioned above, this will be cleaned up as a follow up.